hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeParentRecordHandler.java

```diff
@@ -77,52 +83,88 @@ public void finalizeParentRecord(@NonNull final AccountID payer, @NonNull final
             }
         }
 
-        /* ------------------------- Hbar changes from transaction including staking rewards ------------------------- */
+        // Hbar changes from transaction including staking rewards
         final var hbarChanges = hbarChangesFrom(writableAccountStore);
-        // any hbar changes listed in child records should not be recorded again in parent record, so deduct them.
-        deductChangesFromChildRecords(hbarChanges, context);
+        final var fungibleChanges = fungibleChangesFrom(writableTokenRelStore, tokenStore);
+        final var nftChanges = nftChangesFrom(writableNftStore, tokenStore);
+        if (context.hasChildRecords()) {
+            // All the above changes maps are mutable
+            deductChangesFromChildRecords(context, fungibleChanges, nftChanges, hbarChanges);
+            // In the current system a parent transaction that has child transactions cannot
+            // *itself* cause any net fungible or NFT transfers; fail fast if that happens
+            validateTrue(fungibleChanges.isEmpty(), FAIL_INVALID);
+            validateTrue(nftChanges.isEmpty(), FAIL_INVALID);
```

Comment:
You mean all the changes will be in parent record transferList ?

Reply:
I mean that _if_ the child records include token transfers, then the parent **must** not have any token transfers itself after subtracting all token transfers from child records.

(Since a parent with children is either a contract or schedule operation; neither of which can directly transfer tokens.)